Source/WebKit/WebProcess/GPU/graphics/WebGPU/RemoteBufferProxy.cpp

```diff
@@ -66,33 +65,35 @@ static bool offsetOrSizeExceedsBounds(size_t dataSize, WebCore::WebGPU::Size64 o
 
 auto RemoteBufferProxy::getMappedRange(WebCore::WebGPU::Size64 offset, std::optional<WebCore::WebGPU::Size64> size) -> MappedRange
 {
-    if (m_data.has_value() && m_data->data()) {
-        if (offsetOrSizeExceedsBounds(m_data->size(), offset, size))
-            return { };
-
-        return { m_data->data() + offset, static_cast<size_t>(size.value_or(m_data->size() - offset)) };
-    }
-
     // FIXME: Implement error handling.
     auto sendResult = sendSync(Messages::RemoteBuffer::GetMappedRange(offset, size));
     auto [data] = sendResult.takeReplyOr(std::nullopt);
 
     if (!data || !data->data() || offsetOrSizeExceedsBounds(data->size(), offset, size))
         return { };
 
-    m_data = WTFMove(data);
-    return { m_data->data() + offset, static_cast<size_t>(size.value_or(m_data->size() - offset)) };
+    return { data->data() + offset, static_cast<size_t>(size.value_or(data->size() - offset)) };
```

Comment:
This cannot return offsets into the IPC buffer, that will be destroyed when the call ends.

Reply:
oh that certainly does seem wrong, let me make this use a callback instead as the buffer would still be alive with that approach

Reply:
should be fixed with https://github.com/WebKit/WebKit/pull/28098